src/blocks/packages.rs

```diff
@@ -0,0 +1,370 @@
+#[derive(Deserialize, Debug, Clone, Copy, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum PackageManager {
+    Apt,
+    Pacman,
+    Aur,
+}
+
+pub async fn run(config: &Config, api: &CommonApi) -> Result<()> {
+    let mut config: Config = config.clone();
+
+    let format = config.format.with_default(" $icon $total.eng(w:1) ")?;
+    let format_singular = config
+        .format_singular
+        .with_default(" $icon $total.eng(w:1) ")?;
+    let format_up_to_date = config
+        .format_up_to_date
+        .with_default(" $icon $total.eng(w:1) ")?;
+
+    // If user provide package manager in any of the formats then consider that also
+    macro_rules! any_format_contains {
+        ($name:expr) => {
+            format.contains_key($name)
+                || format_singular.contains_key($name)
+                || format_up_to_date.contains_key($name)
+        };
+    }
+
+    let apt = any_format_contains!("apt");
+    let aur = any_format_contains!("aur");
+    let pacman = any_format_contains!("pacman");
+
+    if !config.package_manager.contains(&PackageManager::Apt) && apt {
+        config.package_manager.push(PackageManager::Apt);
+    }
+    if !config.package_manager.contains(&PackageManager::Pacman) && pacman {
+        config.package_manager.push(PackageManager::Pacman);
+    }
+    if !config.package_manager.contains(&PackageManager::Aur) && aur {
+        config.package_manager.push(PackageManager::Aur);
+    }
+
+    let warning_updates_regex = config
+        .warning_updates_regex
+        .as_deref()
+        .map(Regex::new)
+        .transpose()
+        .error("invalid warning updates regex")?;
+    let critical_updates_regex = config
+        .critical_updates_regex
+        .as_deref()
+        .map(Regex::new)
+        .transpose()
+        .error("invalid critical updates regex")?;
+    let ignore_updates_regex = config
+        .ignore_updates_regex
+        .as_deref()
+        .map(Regex::new)
+        .transpose()
+        .error("invalid ignore updates regex")?;
+
+    // Setup once everything it takes to check updates for every package manager
+    for package_manager in &config.package_manager {
+        let mut backend: Box<dyn Backend> = match package_manager {
+            PackageManager::Apt => Box::new(Apt::new()),
+            PackageManager::Pacman => Box::new(Pacman::new()),
+            PackageManager::Aur => Box::new(Aur::new()),
+        };
+
+        backend.setup().await?;
+    }
+
+    loop {
+        let (mut apt_count, mut pacman_count, mut aur_count) = (0, 0, 0);
+        let mut critical_vec = vec![false];
+        let mut warning_vec = vec![false];
+
+        // Iterate over the all package manager listed in Config
+        for package_manager in config.package_manager.clone() {
+            match package_manager {
+                PackageManager::Apt => {
+                    let mut apt = Apt::new();
+                    apt.ignore_updates_regex = ignore_updates_regex.clone();
```

Comment:
I'm a bit confused why you have this `match` when you could've saved the list of package managers as a `Vec<Box<dyn Backend>>` (you already construct `Box<dyn Backend>` above, but then discard them).